src/https/utilities/https-utilities

```diff
@@ -36,7 +36,7 @@ generate_self_signed_certificate()
 	chmod 750 "$(dirname "$SELF_SIGNED_CHAIN")"
 
 	openssl req -newkey rsa:4096 -nodes -keyout "$SELF_SIGNED_KEY" \
-	        -x509 -days 90 -out "$SELF_SIGNED_CERT" -subj "/O=Nextcloud"
+	        -x509 -days 90 -out "$SELF_SIGNED_CERT" -subj "/O=Nextcloud/CN=Nextcloud"
```

Comment:
You literally only need a CN entry? It doesn't even matter what is is?! This seems broken: the CN is supposed to be the domain name you're using to visit the site protected by the cert.

Reply:
Yes, this fixes it in order to mount webdav drives. For example davfs2 complaints about a missing CN:
```
/sbin/mount.davfs: Mounting failed.
Server certificate was missing commonName attribute in subject name```

The CN value is actually deprecated, browsers use the SAN value. But for some reason, davfs2 checks the CN. I think in order to do it properly, we can fill in the current hostname of the server.

Reply:
I have adjusted the CN to be dynamic now based on the hostname. Feel free to merge or close, it fixed it for our completely internal setup 😄